src/D2L.CodeStyle.Analyzers/ApiUsage/Events/EventHandlerLoaderTypesAnalyzer.cs

```diff
@@ -0,0 +1,181 @@
+				"RegisterEventHandler",
+				"RegisterOrgEventHandler"
+			);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+			Diagnostics.EventTypeMissingEventAttribute,
+			Diagnostics.EventHandlerTypeMissingEventAttribute
+		);
+
+		public override void Initialize( AnalysisContext context ) {
+
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction( RegisterAnalysis );
+		}
+
+		private void RegisterAnalysis( CompilationStartAnalysisContext context ) {
+
+			Compilation compilation = context.Compilation;
+
+			INamedTypeSymbol eventAttributeType = compilation.GetTypeByMetadataName( EventAttributeFullName );
+			if( eventAttributeType.IsNullOrErrorType() ) {
+				return;
+			}
+
+			INamedTypeSymbol eventHandlerAttributeType = compilation.GetTypeByMetadataName( EventHandlerAttributeFullName );
+			if( eventHandlerAttributeType.IsNullOrErrorType() ) {
+				return;
+			}
+
+			IImmutableSet<ISymbol> genericRegisterMethods = GetGenericRegisterMethods( compilation )
+				.ToImmutableHashSet();
+
+			context.RegisterSyntaxNodeAction(
+					ctxt => AnalyzeMethodInvocation(
+						ctxt,
+						(InvocationExpressionSyntax)ctxt.Node,
+						eventAttributeType,
+						eventHandlerAttributeType,
+						genericRegisterMethods
+					),
+					SyntaxKind.InvocationExpression
+				);
+		}
+
+		private void AnalyzeMethodInvocation(
+				SyntaxNodeAnalysisContext context,
+				InvocationExpressionSyntax invocation,
+				INamedTypeSymbol eventAttributeType,
+				INamedTypeSymbol eventHandlerAttributeType,
+				IImmutableSet<ISymbol> genericRegisterMethods
+			) {
+
+			ISymbol expessionSymbol = context.SemanticModel
+				.GetSymbolInfo( invocation.Expression )
+				.Symbol;
+
+			if( expessionSymbol.IsNullOrErrorType() ) {
+				return;
+			}
+
+			if( !( expessionSymbol is IMethodSymbol methodSymbol ) ) {
+				return;
+			}
+
+			if( !methodSymbol.IsGenericMethod ) {
+				return;
+			}
+
+			if( !genericRegisterMethods.Contains( methodSymbol.OriginalDefinition ) ) {
+				return;
+			}
+
+			ITypeSymbol eventTypeSymbol = methodSymbol.TypeArguments[ 0 ];
+			if( eventTypeSymbol.IsNullOrErrorType() ) {
+				return;
+			}
+
+			ITypeSymbol eventHandlerSymbol = methodSymbol.TypeArguments[ 1 ];
+			if( eventHandlerSymbol.IsNullOrErrorType() ) {
+				return;
+			}
+
+			InspectEventType( context, invocation, eventAttributeType, eventTypeSymbol );
+			InspectEventHandlerType( context, invocation, eventHandlerAttributeType, eventHandlerSymbol );
+		}
+
+		private static void InspectEventType(
+				SyntaxNodeAnalysisContext context,
+				InvocationExpressionSyntax invocation,
+				INamedTypeSymbol eventAttributeType,
+				ITypeSymbol eventTypeSymbol
+			) {
+
+			bool hasAttr = eventTypeSymbol
+				.GetAttributes()
+				.Any( attr => attr.AttributeClass.Equals( eventAttributeType ) );
+
+			if( hasAttr ) {
+				return;
+			}
+
+			if( eventTypeSymbol.TypeKind == TypeKind.TypeParameter ) {

```

Comment:
Hmmm... any concern about things getting through via this?

Reply:
Don't think this is need for now.  Removed